cdap-ui/app/cdap/components/DataPrep/Directives/Calculate/index.js

```diff
@@ -0,0 +1,626 @@
+    }
+
+    return (
+      <div>
+        <div className={classnames({"bigger-new-column-label": this.state.operationPopoverOpen === 'CHARCOUNT'})}>
+          {inputLabel}
+        </div>
+        <Input
+          type="text"
+          className="form-control mousetrap"
+          value={this.state.newColumnInput}
+          onChange={this.setNewColumnInput}
+          placeholder={inputPlaceholder}
+          autoFocus
+        />
+        {
+          columnNameAlreadyExists(this.state.newColumnInput) ? (
+            <WarningContainer
+              message={T.translate(`${PREFIX}.newColumnInputDuplicate`)}
+            />
+          ) : null
+        }
+      </div>
+    );
+  }
+
+  renderActionButtons() {
+    return (
+      <div className="action-buttons">
+        <button
+          className="btn btn-primary float-xs-left"
+          disabled={this.isApplyDisabled()}
+          onClick={this.getExpressionAndApply}
+        >
+          {T.translate('features.DataPrep.Directives.apply')}
+        </button>
+
+        <button
+          className="btn btn-link float-xs-right"
+          onClick={this.setDefaultState}
+        >
+          {T.translate('features.DataPrep.Directives.cancel')}
+        </button>
+      </div>
+    );
+  }
+
+  renderOperationPopover() {
+    if (!this.state.operationPopoverOpen) { return null; }
+
+    if (this.state.operationPopoverOpen === 'CHARCOUNT') {
+      return (
+        <div
+          className="third-level-popover"
+          onClick={preventPropagation}
+        >
+          {this.renderNewColumnNameInput()}
+
+          <hr />
+
+          {this.renderActionButtons()}
+        </div>
+      );
+    }
+
+    let operationSign;
+    let inputMin;
+
+    if (this.state.operationPopoverOpen === 'MODULO') {
+      inputMin = 1;
+    }
+
+    switch (this.state.operationPopoverOpen) {
+      case 'ADD':
+        operationSign = '+';
+        break;
+      case 'SUBTRACT':
+        operationSign = '-';
+        break;
+      case 'MULTIPLY':
+        operationSign = 'x';
+        break;
+      case 'DIVIDE':
+        operationSign = '/';
+        break;
+      case 'MODULO':
+        operationSign = '%';
+        break;
+    }
+
+    return (
+      <div
+        className="third-level-popover"
+        onClick={preventPropagation}
+      >
+        {
+          this.state.operationPopoverOpen === 'POWEROF' ?
+            (
+              <div className="operation-description">
+                {T.translate(`${PREFIX}.Options.POWEROF.description`)}
+              </div>
+            )
+          : null
+        }
+        <div className="operation-sign-input">
+          <span className="operation-sign">
+            {operationSign}
+          </span>
+          <Input
+            className="operation-input mousetrap"
+            value={this.state.operationInput}
+            onChange={this.setOperationInput}
```

Comment:
This will still allow the user to type `-20` in the number input textbox. Lets fix this in another PR as its not crucial (no one will enter -20 for a modulo operation in reality).